dist/src/main/java/io/camunda/tasklist/zeebe/ServiceTaskMutationAdapter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.tasklist.zeebe;
+
+import io.camunda.authentication.tenant.TenantAttributeHolder;
+import io.camunda.security.auth.Authentication;
+import io.camunda.service.JobServices;
+import io.camunda.service.UserTaskServices;
+import io.camunda.tasklist.webapp.service.TaskMutationAdapter;
+import io.camunda.webapps.schema.entities.tasklist.TaskEntity;
+import io.camunda.zeebe.gateway.rest.RequestMapper;
+import java.util.List;
+import java.util.Map;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("!standalone")
```

Comment:
I am wondering why this cannot be generalized to all profiles.
I think it just requires https://github.com/camunda/camunda/blob/main/dist/src/main/java/io/camunda/application/commons/broker/client/BrokerClientConfiguration.java which is always created as it is part of `io.camunda.application.commons` package

Reply:
In that iteration, I just wanted to differentiate between starting Tasklist standalone with `StandaloneTasklist` and as a single application with `StandaloneCamunda`. My thinking is that if a user continues deploying the old architecture with 8.7, i.e., they deploy Tasklist standalone, which gets started with `StandaloneTasklist`, then "nothing" should change in terms of how Tasklist communicates with the Gateway/Process Engine. If they configured Zeebe Client credentials and assigned specific Tenants to the Zeebe Client, then nothing should be changed here. Basically, I was thinking of keeping a level of "backward compatibility".

If running in a single application or moving to a new architecture, the communication can happen via the services (eventually via the broker clients).

That's basically the direction where I am coming from.

Reply:
It makes sense. I was wondering if in a next iteration we can still bypass the REST and gRPC calls even in standalone mode. For example, it can only be activated with some configuration.